### PR TITLE
Add Missing Additional Server Configuration Documentation

### PR DESCRIPTION
**Pull request by settermjd**
This commit adds additional server configuration documentation from the `ios_universal_links` branch, that is missing from the master-antora branch. It was introduced in cd99beb054fbd8bef9e30948cf75e046cfdc9cd6.

💁‍♂️ Do not merge this PR before #33 is merged.

### CODE REVIEW

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+. Create {apple-app-site-association-file-url}[an apple-app-site-association file] that contains JSON data about the URLs that your app can handle.
+. Upload the apple-app-site-association file to your HTTPS web server. You can place the file at the root of your server or in the `.well-known` subdirectory.
+. Prepare your app to handle universal links. 
+____
+
+The `apple-app-site-association` will be generated by ownbrander, and needs to be served statically on your site through HTTPS. 
+You can safely place it on the root folder of your ownCloud installation (e.g., `/var/www/owncloud`). 
+To achieve the second requirement, some changes will also need to be completed on your Apache configuration.
+
+If you configured your installation with the official {admin_manual}[Admin Manual], your `owncloud.conf` file should include the following:
```
Comment (mmattel) [inline modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: I propose to add a `=== Apache` section and below a `=== nginx` section.
I will add on a second comment the nginx info.

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+____
+Adding support for universal links is easy. 
+There are three steps you need to take:
+
+. Create {apple-app-site-association-file-url}[an apple-app-site-association file] that contains JSON data about the URLs that your app can handle.
+. Upload the apple-app-site-association file to your HTTPS web server. You can place the file at the root of your server or in the `.well-known` subdirectory.
+. Prepare your app to handle universal links. 
+____
+
+The `apple-app-site-association` will be generated by ownbrander, and needs to be served statically on your site through HTTPS.
```
Comment (mmattel) [inline modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: The `apple-app-site-association`...
Not clear what this is. Is this a link, a file, some content?
When generated by ownbrander, where does it get written to? configurable? needs to be copied ?...

Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: Fair call. I'll find out more.

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc (hunk 1 above)
Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: I've made those changes. Just waiting on your NGINX configuration, and I'll add that in as well. Unless you're adding that yourself, locally.

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc (hunk 2 above)
Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: I linked to the ownBrander section of the documentation. That section I'm not too familiar with, but I would expect regular users to be.

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+____
+Adding support for universal links is easy. 
+There are three steps you need to take:
+
+. Create {apple-app-site-association-file-url}[an apple-app-site-association file] that contains JSON data about the URLs that your app can handle.
+. Upload the apple-app-site-association file to your HTTPS web server. You can place the file at the root of your server or in the `.well-known` subdirectory.
+. Prepare your app to handle universal links. 
+____
+
+The `apple-app-site-association` will be generated by xref:branded_ios_app/publishing_ios_app_7.adoc#generate-ios-app[ownBrander], and needs to be served statically on your site through HTTPS.
```
Comment (mmattel) [inline modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: ...
The `apple-app-site-association` is a subdirectory of your owncloud URL. Data generated by xref:branded_ios_app/publishing_ios_app_7.adoc#generate-ios-app[ownBrander] will be accessed via this subdirectory and needs to be served statically on your site through HTTPS. `apple-app-site-association` can be either a direct subdirectory or `/.well-known/apple-app-site-association`. The name `apple-app-site-association` is given and must not be changed. The file which gets accessed when using this subdirectory is also named `apple-app-site-association` without any extension.
When this subdirectory is accessed, the webserver must be instructed that the content type served is defined as `'application/json'`.
The physical path which is used when accessing this directory must be defined in your webserver config.
NOTE: Please consider when using a physical path to the file inside your owncloud directory, that this file must be present when you upgrade ownCloud or you chose a different path outside owncloud.
...

I hope this makes it more clear what it is about.
@settermjd pls improve/reword if needed.

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+=== Apache Configuration
+
+To achieve the second requirement, some changes will also need to be completed on your Apache configuration.
+
+If you configured your installation with the official {admin_manual}[Admin Manual], your Apache `owncloud.conf` file should include the following:
+
+[source,apacheconf]
+....
+# Create two alias for the file (for compatibility reasons):
+AliasMatch "^/(\.well-known/)?apple-app-site-association$" "/var/www/owncloud/apple-app-site-association"
```
Comment (mmattel) [inline modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: This is only one alias, the second one is missing.
I guess that the missing alias is:
`AliasMatch "/owncloud/apple-app-site-association" "/var/www/owncloud/apple-app-site-association"`
Then both the `.well-known/apple-app-site-association` and `apple-app-site-association` subdirs are defined.

pls see the definition and difference of AliasMatch for more details
https://httpd.apache.org/docs/2.4/mod/mod_alias.html#aliasmatch

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Also, a new `RewriteCond` directive, included in the code block below, needs to be included in your `.htaccess`, so that no other redirections will apply to any of these two paths.
+
+[source,apacheconf]
+....
+RewriteCond %\{REQUEST_URI} !^/(.well-known/)?apple-app-site-association$
+....
+
+=== NGINX Configuration
+
```
Comment (mmattel) [inline modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: For nginx, the directives are:
```
  location = /.well-known/apple-app-site-association {
    root '/var/www/owncloud';
    default_type 'application/json';
  }
  location = /apple-app-site-association {
    root '/var/www/owncloud';
    default_type 'application/json';
  }
```

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc (hunk 3 above)
Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: So this text is all that's required? I didn't touch the text much, as I know very little about the subject matter.

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc (hunk 4 above)
Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: Can you review the revised text to ensure that I've understood you correctly, pls?

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc (hunk 3 above)
Comment (mmattel) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: That text should be added, pls check for writung improvements.

Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: I've added it in and made some adjustments. However, it feels rough to me. Can you review pls, @mmattel?

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+____
+Adding support for universal links is easy. 
+There are three steps you need to take:
+
+. Create {apple-app-site-association-file-url}[an apple-app-site-association file] that contains JSON data about the URLs that your app can handle.
+. Upload the apple-app-site-association file to your HTTPS web server. You can place the file at the root of your server or in the `.well-known` subdirectory.
+. Prepare your app to handle universal links. 
+____
+
+The `apple-app-site-association` directory is generated by xref:branded_ios_app/publishing_ios_app_7.adoc#generate-ios-app[ownBrander] and must be served statically over HTTPS.
```
Comment (mmattel) [inline modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: The `apple-app-site-association` directory

From that what I read and understood, it is NOT a directory that is generated by ownBrander but a data file.
Pls also see the statement in line 35. (yes a bit confusing but so it works...)

Therefore I propose to write:
The `apple-app-site-association` data

Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: Agreed.

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc [hunk 7]
```diff
@@ -0,0 +1,88 @@
+= Additional Server Configuration 
+:apple-universal-link-support-url: https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/UniversalLinks.html
```
Comment (mmattel) [inline modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: Missing a `:toc: right`

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+:admin_manual: https://doc.owncloud.com/server/administration_manual/index.html
+:apple-app-site-association-file-url: https://developer.apple.com/documentation/security/password_autofill/setting_up_an_app_s_associated_domains
+:aliasmatch-uri: https://httpd.apache.org/docs/2.4/mod/mod_alias.html#aliasmatch
+:rewritecond-directive-uri: http://httpd.apache.org/docs/2.2/mod/mod_rewrite.html#RewriteCond
+:generate-ios-app-url: branded_ios_app/publishing_ios_app_7.adoc#generate-ios-app
+
+== Add Support for Apple Universal Links
```
Comment (mmattel) [inline modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: Proposal, add:
```
== Introduction

Support of Universal Links
When you support universal links, iOS users can tap a link to your website and get seamlessly redirected to
your  installed app without going through Safari. If your app isn’t installed, tapping a link to your website
opens your website in Safari. For more details see:
https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html
```

**Comment by mmattel** [comment]
Info: #33 is merged already, therefore no longer blocking point

File: modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc (hunk 8 above)
Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: I'm not an iOS developer or guru, but this seems a little strange to me. Did you mean this as an admonition or aspect of supporting documentation for the "Add Support for Apple Universal Links" section on that page?

Comment (mmattel) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: yes, aspect of supporting documentation. helpful to understand the topic

Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: Thanks.

Comment (settermjd) [reply modules/ROOT/pages/branded_ios_app/publishing_ios_app_10.adoc]: Done.

**Pull request merged**